### ISSUE TITLE
Question about manual data update

### ISSUE THREAD

**Issue by claustres** [perliedman/leaflet-realtime#136]
I wonder why in the manual update case the `removeMissing` option is not taken into account [here](https://github.com/perliedman/leaflet-realtime/blob/master/src/L.Realtime.js#L85) ?

Indeed it is very convenient to let the plugin manages this for you in most use cases instead of having to manually split features between added/removed ones. If you still want to use this manual workflow simply set the  `removeMissing` option to `false`.

This way it seems to me the API is more coherent to what is done when data fetching is delegated to the plugin [here](https://github.com/perliedman/leaflet-realtime/blob/master/src/L.Realtime.js#L87). However, I also understand this is a breaking change to endorse...

**Comment by perliedman** [comment]
Hi,

I had to dig through the history to figure out what the reasoning was!

Anyway, the `update` method was initially created for the use case when updates are pushed from the server using websockets or similar, rather than being polled. For this use case, individual updates will be sent for each feature, not all features at the same time. So the `update` method is for _updating_ (and potentially adding) one or more features, but not removing features. If features would have been auto-removed, updating a single feature would at the same time remove all other features from the map.

Having said that, I can see that it would potentially be useful to be able to also remove features for other use cases. Perhaps we could add an options parameter where you could also specify if items should be removed (defaulting to `false`).

**Comment by claustres** [comment]
Thanks I now understand the main use case, however it is only relevant to push individual features only if they move slowly and not in sync. We have some use cases where all features "move" (ie added/updated/removed) eg each 5 seconds and in this case it is better to send a whole update (ie "batch") with all current features rather than individual features, notably when the feature number is large (like thousands)

**Comment by perliedman** [comment]
Ok, yes I understand your use case. If you are up for it, I would be happy to look at a PR along the lines I mentioned above, adding an option to override the default behavior.

**Comment by claustres** [comment]
Well I was thinking to simply replace [this line](https://github.com/perliedman/leaflet-realtime/blob/master/src/L.Realtime.js#L85) by `this._onNewData(this.options.removeMissing, geojson)` since the `_onNewData()` method already provide the flag and this will be similar than with auto-updates.

However this is a breaking change, so we might add a flag in the update method as you seem to propose and keep the old behavior like this:
```js
update: function(geojson, removeMissing) {
    ...
    if (geojson) {
        this._onNewData(removeMissing, geojson); // Defaults to false if option not defined
    } else {
        responseHandler = L.bind(function(data) {
            // Will allow to temporarily override layer option if provided
            this._onNewData(typeof removeMissing === 'undefined' ? this.options.removeMissing : removeMissing, data);
        }, this);
        ...
    }

    return this;
}
```

What do you think ?

**Comment by PeterPan73** [comment]
Hi @claustres, I am also trying to manually update the data with _realtime.update()_ and socket, but _removeMissing_ not working. As I can see from the comment above from @perliedman _update_ only add and update data. Is there any solution to this without _remove_?

**Comment by claustres** [comment]
Please clarify your use case because at some point you will probably need to remove "dead" features anyway. This plugin assumes that updates are not performed in a single batch but incrementally so it does not automatically remove features disappearing between two updates, for this you will need to call the remove method manually, probably by listening to some `remove` event on your socket. I was suggesting to support the `removeMissing` option for use cases where the update is performed as a single operation/batch with all features and in this case it would be relevant to automatically remove features disappearing between two updates.

**Comment by PeterPan73** [comment]
Hi @claustres thanks for the reply.
I use socket to update the data with `socket.on("updateShips", function(myJSON) {
			realtime.update(myJSON);
		});` witch update every 1 second.
The problem is when a marker is no longer in the data, it remains on the map and is not removed. I just cant find how to remove the marker if he is not in the data. I also use _updateFeature_ to update icon base on status. If there is another option for adding, updating, and deleting markers, I will be  happy if you share an idea. Thanks.

**Comment by claustres** [comment]
You can bypass this behavior using `layer._onNewData(true, geoJson)` I guess, it will remove the missing feature.

**Comment by PeterPan73** [comment]
@claustres where to put this?

**Comment by PeterPan73** [comment]
> Well I was thinking to simply replace [this line](https://github.com/perliedman/leaflet-realtime/blob/master/src/L.Realtime.js#L85) by `this._onNewData(this.options.removeMissing, geojson)` since the `_onNewData()` method already provide the flag and this will be similar than with auto-updates.
> 
> However this is a breaking change, so we might add a flag in the update method as you seem to propose and keep the old behavior like this:
> 
> ```js
> update: function(geojson, removeMissing) {
>     ...
>     if (geojson) {
>         this._onNewData(removeMissing, geojson); // Defaults to false if option not defined
>     } else {
>         responseHandler = L.bind(function(data) {
>             // Will allow to temporarily override layer option if provided
>             this._onNewData(typeof removeMissing === 'undefined' ? this.options.removeMissing : removeMissing, data);
>         }, this);
>         ...
>     }
> 
>     return this;
> }
> ```
> 
> What do you think ?

@perliedman I think this is great update, I hope add in next update. @claustres thanks again!

**Comment by perliedman** [comment]
Closing this since I believe it is fixed by #127.
